AutoFollowIT#testPauseAndResumeWithMultipleAutoFollowPatterns failing

This PR build failed: https://gradle-enterprise.elastic.co/s/ercrostvlu5k2/console-log?task=:x-pack:plugin:ccr:internalClusterTest

```
org.elasticsearch.xpack.ccr.AutoFollowIT > testPauseAndResumeWithMultipleAutoFollowPatterns FAILED
    java.lang.AssertionError: 
    Expected: a value equal to or greater than <9L>
         but: <8L> was less than <9L>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ccr.AutoFollowIT.lambda$testPauseAndResumeWithMultipleAutoFollowPatterns$33(AutoFollowIT.java:533)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:874)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:847)
        at org.elasticsearch.xpack.ccr.AutoFollowIT.testPauseAndResumeWithMultipleAutoFollowPatterns(AutoFollowIT.java:533)

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=487, name=Thread-4, state=RUNNABLE, group=TGRP-AutoFollowIT]

        Caused by:
        java.lang.AssertionError
            at __randomizedtesting.SeedInfo.seed([2F4CFB3BB06F1141]:0)
            at org.junit.Assert.fail(Assert.java:86)
            at org.junit.Assert.assertTrue(Assert.java:41)
            at org.junit.Assert.assertTrue(Assert.java:52)
            at org.elasticsearch.xpack.ccr.AutoFollowIT.lambda$testPauseAndResumeWithMultipleAutoFollowPatterns$22(AutoFollowIT.java:503)
            at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:874)
            at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:847)
            at org.elasticsearch.xpack.ccr.AutoFollowIT.lambda$testPauseAndResumeWithMultipleAutoFollowPatterns$23(AutoFollowIT.java:503)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:ccr:internalClusterTest' --tests "org.elasticsearch.xpack.ccr.AutoFollowIT.testPauseAndResumeWithMultipleAutoFollowPatterns" -Dtests.seed=2F4CFB3BB06F1141 -Dtests.security.manager=true -Dtests.locale=ar-EG -Dtests.timezone=Africa/Porto-Novo -Dcompiler.java=12 -Druntime.java=11
```

Build stats shows other occasional failures of this test recently. The `REPRODUCE WITH` line does not reproduce it for me, but running the test in a loop did yield a failure within a few hundred attempts. There seems to be another (possibly-related) failure of this form too:

```
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:ccr:internalClusterTest' --tests "org.elasticsearch.xpack.ccr.AutoFollowIT.testPauseAndResumeWithMultipleAutoFollowPatterns {seed=[32D8D58CB0002449:59EC7D8F7D331D14]}" -Dtests.seed=32D8D58CB0002449 -Dtests.security.manager=true -Dtests.jvms=4 -Dtests.locale=be-BY -Dtests.timezone=America/Fort_Nelson -Dcompiler.java=12 -Druntime.java=12
  2> java.lang.AssertionError: 
    Expected: <20>
         but: was <19>
        at __randomizedtesting.SeedInfo.seed([32D8D58CB0002449:59EC7D8F7D331D14]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ccr.AutoFollowIT.testPauseAndResumeWithMultipleAutoFollowPatterns(AutoFollowIT.java:552)
```

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
@@ -491,10 +491,9 @@ public class AutoFollowIT extends CcrIntegTestCase {
 
         // start creating new indices on the remote cluster
         final Thread createNewLeaderIndicesThread = new Thread(() -> {
-            int leaderIndicesCount;
-            while (running.get() && (leaderIndicesCount = leaderIndices.incrementAndGet()) < 20) {
+            while (running.get() && leaderIndices.get() < 20) {
                 final String prefix = randomFrom(prefixes);
-                final String leaderIndex = prefix + leaderIndicesCount;
+                final String leaderIndex = prefix + leaderIndices.incrementAndGet();
                 try {
                     createLeaderIndex(leaderIndex, leaderIndexSettings);
                     ensureLeaderGreen(leaderIndex);

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
@@ -527,7 +527,8 @@ public class AutoFollowIT extends CcrIntegTestCase {
 
         // wait for more leader indices to be created on the remote cluster
         assertBusy(() -> assertThat(leaderIndices.get(), greaterThanOrEqualTo(9)));
-        assertBusy(() -> assertThat(getAutoFollowStats().getNumberOfSuccessfulFollowIndices(), greaterThanOrEqualTo(9L)));
+        assertBusy(() -> assertThat(getAutoFollowStats().getNumberOfSuccessfulFollowIndices(), greaterThanOrEqualTo(9L)),
+            30L, TimeUnit.SECONDS);
 
         running.set(false);
         createNewLeaderIndicesThread.join();

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
@@ -520,7 +520,6 @@ public class AutoFollowIT extends CcrIntegTestCase {
 
         // wait for more leader indices to be created on the remote cluster
         assertBusy(() -> assertThat(leaderIndices.get(), greaterThanOrEqualTo(6)));
-        assertBusy(() -> assertThat(getAutoFollowStats().getNumberOfSuccessfulFollowIndices(), greaterThanOrEqualTo(6L)));
 
         // resume auto follow patterns
         pausedAutoFollowerPatterns.forEach(this::resumeAutoFollowPattern);

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java
@@ -498,8 +498,7 @@ public class AutoFollowIT extends CcrIntegTestCase {
                     createLeaderIndex(leaderIndex, leaderIndexSettings);
                     ensureLeaderGreen(leaderIndex);
                     if (pausedAutoFollowerPatterns.stream().noneMatch(pattern -> pattern.startsWith(prefix))) {
-                        final String followingIndex = "copy-" + leaderIndex;
-                        assertBusy(() -> assertTrue(ESIntegTestCase.indexExists(followingIndex, followerClient())));
+                        ensureFollowerGreen("copy-" + leaderIndex);
                     } else {
                         Thread.sleep(200L);
                     }

